kobo/apps/django_allauth/admin.py

```diff
@@ -0,0 +1,48 @@
+from allauth.socialaccount.admin import SocialAppForm, SocialAppAdmin
+from allauth.socialaccount.models import SocialApp
+from django.contrib import admin
+from django.core.exceptions import ValidationError
+from django.db.models import Q
+
+
+class RequireProviderIdSocialAppForm(SocialAppForm):
+    def __init__(self, *args, **kwargs):
+        super(SocialAppForm, self).__init__(*args, **kwargs)
+        # require the provider_id in the admin, since we can't make it required on allauth's model
+        self.fields['provider_id'].required = True
+
+    def clean_provider_id(self):
+        reserved_keywords = ['kobo']
```

Comment:
IMO if we are blocking in django admin, it would be fine to remove the "Please do not use `kobo` as the `id` value" check. But I'll defer to you.

Reply:
Good point, especially since the code to parse OIDC sub-providers has been removed - it's now guarding against an impossible-to-achieve state (at least for someone not trying to break it.)

Reply:
Actually, nevermind - we still need `check_socialaccount_providers`, because we also need to check the `id` of any social accounts, and that gets set on an env var.